lib/core/test.js

```diff
@@ -86,7 +89,11 @@ Test.prototype.deployAll = function(contractsConfig, cb) {
       }
       self.web3.eth.defaultAccount = accounts[0];
       RunCode.doEval(result, self.web3); // jshint ignore:line
-      cb();
+      if (self.options.logging) {
+        cb(logger);
```

Comment:
Now that I think about this one, your approach is probably better.

The thing that changed my mind is that I realized I'm sometimes directly passing `mocha`'s `done` parameter into `deployAll`, which takes the presence of an argument as an error condition.  If the callback is being called with `logger`, and the callback is `done` (as is in some boilerplate, etc), it will be interpreted by mocha as an error condition.